.github/workflows/ci-backend.yml

```diff
@@ -118,6 +115,61 @@ jobs:
                   CLICKHOUSE_HOST: 'localhost'
                   CLICKHOUSE_SECURE: 'False'
                   CLICKHOUSE_VERIFY: 'False'
+              run: |
+                  mkdir -p frontend/dist
+                  touch frontend/dist/index.html
+                  touch frontend/dist/layout.html
+                  touch frontend/dist/shared_dashboard.html
+                  pytest ee -m "not saml_only"
+
+    saml:
+        name: Django tests – with SAML
+        runs-on: ubuntu-latest
+
+        steps:
+            - uses: actions/checkout@v1
+              with:
+                  fetch-depth: 1
+
+            - name: Start stack with Docker Compose
+              run: docker-compose -f ee/docker-compose.ch.yml up -d db clickhouse
+
+            - name: Set up Python
+              uses: actions/setup-python@v2
+              with:
+                  python-version: 3.9
+
+            - name: Install SAML (python3-saml) dependencies (not required for Cloud or FOSS)
+              run: sudo apt-get install libxml2-dev libxmlsec1-dev libxmlsec1-openssl
+
+            - uses: syphar/restore-virtualenv@v1
+              id: cache-backend-tests
+            - uses: syphar/restore-pip-download-cache@v1
+              if: steps.cache-backend-tests.outputs.cache-hit != 'true'
+            - name: Install python dependencies
+              if: steps.cache-backend-tests.outputs.cache-hit != 'true'
+              run: |
+                  python -m pip install -r requirements-dev.txt
+                  python -m pip install -r requirements.txt
+                  python -m pip install python3-saml==1.12.0
+
+            - name: Run posthog tests
+              env:
+                  DATABASE_URL: 'postgres://posthog:posthog@localhost:5432/posthog'
+              run: |
+                  mkdir -p frontend/dist
+                  touch frontend/dist/index.html
+                  touch frontend/dist/layout.html
+                  touch frontend/dist/shared_dashboard.html
+                  pytest posthog/
```

Comment:
Rather than running everything, worth running only things with some specific pytest.mark?

Reply:
Not sure if it's still worth running all tests as this is the only context where we run them with all the extra SAML dependencies, making sure that having them doesn't break something down.